rct229/rules/section5.py

```diff
@@ -97,3 +99,88 @@ def rule_check(self, context, calc_vals, data=None):
 
 
 # ------------------------
+
+
+class Section5Rule44(RuleDefinitionListIndexedBase):
+    """Rule 44 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule44, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section5Rule44.BuildingRule(),
+            index_rmr="baseline",
+            id="5-44",
+            description="The infiltration modeling method in the baseline includes adjustment for weather and building operation.",
+            rmr_context="buildings",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule44.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={"$..infiltration[*]": ["modeling_method"]},
+            )
+
+        def get_calc_vals(self, context, data=None):
+            baseline_infiltration = find_all("$..infiltration[*]", context.baseline)
+            failing_infiltration_ids = [
+                b_infiltration["id"]
+                for b_infiltration in baseline_infiltration
+                if b_infiltration["modeling_method"] != CONSTANT
+            ]
+            return {"failing_infiltration_ids": failing_infiltration_ids}
+
+        def rule_check(self, context, calc_vals, data=None):
+            return len(calc_vals["failing_infiltration_ids"]) == 0
+
+
+# ------------------------
+
+
+class Section5Rule46(RuleDefinitionListIndexedBase):
+    """Rule 44 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule46, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule46.BuildingRule(),
+            index_rmr="proposed",
+            id="5-46",
+            description="The infiltration shall be modeled using the same methodology and adjustments for weather and building operation in both the proposed design and the baseline building design.",
+            rmr_context="buildings",
+        )
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule46.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True)
+            )
+
+        def get_calc_vals(self, context, data=None):
+            failing_infiltration_zone_ids = []
+            baseline_zones = find_all("$..zones[*]", context.baseline)
+            proposed_zones = find_all("$..zones[*]", context.proposed)
+
+            # This assumes that the surfaces all match
+            matched_baseline_zones = match_lists_exactly_by_id(
```

Comment:
I'm thinking that it might be better to use `match_lists_by_id()` here and treat any missing baseline matches as failures. Otherwise, any missing baseline matches will cause this code to throw an exception.

Reply:
In this case, I can add a failure check after this line:

if len (matched_baseline_zones) != len(proposed_zones):
   # Search for the missing zones in the proposed_zone list
   # Add the missing zones to the failed list for report
   return {"failed_zones" : failed_zone_list}

proposed_baseline_zone_pairs = zip(proposed_zones, matched_baseline_zones)
.... continue the code.

Reply:
I think you will still need to check for b_zone is None inside your for loop.

Here is an alternate idea that avoids match_lists() all together. Starting after line 162:
```
b_zone_dict = {b_zone["id"]: b_zone for b_zone in baseline_zones}. # Hurray for dict comprehensions!

for p_zone in proposed_zones:
    # Get matching baseline zone
    b_zone = b_zone_dict.get(p_zone["id"]) # returns None if the id is not a key in b_zone_dict
    
    if b_zone is None:
      failing_infiltration_zone_ids.append(p_zone["id"])
    else:
       ...
```

Reply:
Done